### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by lisadejonghe**
This pull request has:

- Refactoring of create projects
- Fix infinite loading of coaches
- Disable color picker for milestone 2
- fix build issues

+ #254

### CODE REVIEW

File: fe/src/features/projects/components/ProjectRoles.vue [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    const errorMessageRoleAmount = ref('')
+
+    return {
+      skillStore,
+      filterRoles,
+      newRolePrompt,
+      newRole,
+      newRoleColor,
+      errorRoleAmount,
+      errorMessageRoleAmount
```
Comment (Zelzahn) [inline fe/src/features/projects/components/ProjectRoles.vue:90]: Is it possible that there's no usage of these 2 variables?

**Review by Zelzahn — CHANGES_REQUESTED** [review]

File: fe/src/features/projects/components/ProjectRoles.vue (hunk 1 above)
Comment (lisadejonghe) [reply fe/src/features/projects/components/ProjectRoles.vue]: Should indeed be removed from this file.

File: fe/src/features/projects/components/subcomponents/NewRoleDialog.vue [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+        label="Role name"
+        lazy-rules
+        :rules="[
+            (val) =>
+              (val && val.length > 0) || 'Enter the name of the new role.',
+          ]"
+      />
+    </q-card-section>
+    <q-card-section class="q-pt-none">
+      <!--        TODO REMOVE -->
```
[the remark this answers is not in the document]
Comment (lisadejonghe) [reply fe/src/features/projects/components/subcomponents/NewRoleDialog.vue:24]: I also wondered if that should be there @lasaeren.

File: fe/src/features/projects/components/subcomponents/DeleteRoleDialog.vue [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    return {
+      skillStore
+    }
+  },
+  methods: {
+    deleteRoleConfirm(id: number) {
+      if (id !== -1) {
+        this.skillStore.deleteSkill(id).then(() => {
+
+          // this does not work if
```
[the remark this answers is not in the document]
Comment (lasaeren) [reply fe/src/features/projects/components/subcomponents/DeleteRoleDialog.vue]: fixed

File: fe/src/features/projects/components/subcomponents/NewRoleDialog.vue (hunk 2 above)
[the remark this answers is not in the document]
Comment (lasaeren) [reply fe/src/features/projects/components/subcomponents/NewRoleDialog.vue:24]: fixed

**Review by lasaeren — APPROVED** [review]

**Pull request merged**